lib/cmsis_rtos_v1/cmsis_mempool.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel_structs.h>
+#include <cmsis_os.h>
+
+#define TIME_OUT	100
+
+/**
+ * @brief Create and Initialize a memory pool.
+ */
+osPoolId osPoolCreate(const osPoolDef_t *pool_def)
+{
+	if (_is_in_isr()) {
+		return NULL;
+	}
+
+	return (osPoolId)pool_def;
+}
+
+/**
+ * @brief Allocate a memory block from a memory pool.
+ */
+void *osPoolAlloc(osPoolId pool_id)
+{
+	osPoolDef_t *osPool = (osPoolDef_t *)pool_id;
+	void *ptr;
+
+	if (k_mem_slab_alloc((struct k_mem_slab *)(osPool->pool),
+				&ptr, TIME_OUT) == 0) {
+		return ptr;
+	} else {
+		return NULL;
+	}
+}
+
+/**
+ * @brief Allocate a memory block from a memory pool and set it to zero.
+ */
+void *osPoolCAlloc(osPoolId pool_id)
+{
+	osPoolDef_t *osPool = (osPoolDef_t *)pool_id;
+	void *ptr;
+
+	if (k_mem_slab_alloc((struct k_mem_slab *)(osPool->pool),
+				&ptr, TIME_OUT) == 0) {
+		memset(ptr, 0, osPool->item_sz);
+		return ptr;
+	} else {
+		return NULL;
+	}
+
+	return ptr;
```

Comment:
@rgundi : Do we need to return? I think we don't reach here, there is already return in both if and else case. Or did I miss understanding anything?

Reply:
You are right. That return statement is not required. I'll correct that. Wonder why the compiler didn't catch this though. It should have said "unreachable statement" or something like that. @nashif Any idea why some warnings are apparently not enabled?

Reply:
we do not have such a warning, just remove it.

Reply:
ok. will remove.